vertica_python/vertica/cursor.py

```diff
@@ -58,8 +58,7 @@ def execute(self, operation, parameters=None):
 
         self.rowcount = 0
         if self.last_execution:
-            # ToDo: can just empty the message buffer in connection if easy to do
-            self.connection.reset_connection()
```

Comment:
I think we should also move logic in connection.process_message to cursor.process_message and delegate all the message processing to that. 

Keeping fetchone light weight is a good idea, as I mentioned we have the logic to process message spread over different places. Better to do a rewrite to put them all in one place.